ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh

```diff
@@ -0,0 +1,363 @@
+ |_____|_| \_|____/  |____/|_____|____/ \___/ \____|
+
+####################################################
+
+FOOTER
+}
+
+print_debug_on_error() {
+  e=$?
+  [ $e -ne 0 ] && print_debug
+  exit $e
+}
+
+log "Applying MTU offset ${MTU_OFFSET} to the cluster"
+
+trap "print_debug_on_error" EXIT
+
+setup_packet_cluster
+
+wait_for_co "600s"
+
+print_debug
+
+log "Getting the current cluster network MTU..."
+cluster_mtu=$(oc get network.config --output=jsonpath='{.items..status.clusterNetworkMTU}')
+if [ -z "${cluster_mtu}" ]; then
+  log "Unable to get clusterNetworkMTU"
+  exit 1
+fi
+log "Cluster network MTU is ${cluster_mtu}"
+
+# Get host default gateway interface & MTU...
+node=$(oc get nodes -o jsonpath='{.items[0].metadata.name}')
+get_node_host_data "${node}"
+
+if [[ ${MTU_OFFSET} -ne 0 ]]; then
+  from=${cluster_mtu}
+  to=$((from+MTU_OFFSET))
+  host_to=$((host_mtu+MTU_OFFSET))
+  oc patch Network.operator.openshift.io cluster --type='merge'   --patch '{"spec":{"migration":null}}'
+  wait_for_cno_cleanup "60s"
+  log "Starting MTU migration from cluster network MTU ${from} and host MTU ${host_mtu} to cluster network MTU ${to} and host MTU ${host_to}"
+  oc patch Network.operator.openshift.io cluster --type='merge' --patch "{\"spec\": { \"migration\": { \"mtu\": { \"network\": { \"from\": ${from}, \"to\": ${to} } , \"machine\": { \"to\" : ${host_to}} } } } }"
+else
+  network_type=$(oc get network.config --output=jsonpath='{.items..status.networkType}')
+  if [ -z "${network_type}" ]; then
+    log "Unable to get networkType"
+    exit 1
+  fi
+  network_config="ovnKubernetesConfig"
+  if [ "${network_type}" = "OpenShiftSDN" ]; then
+    network_config="openshiftSDNConfig"
+  fi
+
+  host_to=$(oc get network.config --output=jsonpath='{.items..status.migration.mtu.machine.to}')
+  if [ -z "${host_to}" ]; then
+    log "Unable to get migration.host.to"
+    exit 1
+  fi
+
+  log "Ending MTU migration to cluster network MTU ${cluster_mtu}"
+  oc patch MachineConfigPool master --type='merge' --patch '{ "spec": { "paused": true } }'
+  oc patch MachineConfigPool worker --type='merge' --patch '{ "spec":{ "paused" :true } }'
+  # we have observed that if we don't space out these updates, MCO will miss one
+  # of them until after some minutes
+  patch_host_mtu "${host_to}"
+  sleep "30s"
+  oc patch Network.operator.openshift.io cluster --type=merge --patch "{ \"spec\": { \"migration\": null, \"defaultNetwork\":{ \"${network_config}\":{ \"mtu\":${cluster_mtu} }}}}"
+  sleep "30s"
+  wait_for_final_mc "900s"
+  oc patch MachineConfigPool master --type='merge' --patch '{ "spec": { "paused": false } }'
+  oc patch MachineConfigPool worker --type='merge' --patch '{ "spec":{ "paused" :false } }'
+fi
+
+# Check all machine config pools are updated
+wait_for_mcp "2700s"
+
+# Check all cluster operators are operational
+wait_for_co "600s"
+
+# Check for some time that there are not firing api alerts
+check_no_api_alerts "60s"
+
```

Comment:
maybe I missed it, but are we validating the correct MTU is there after the patch is done to change it? Also, wondering
how important it might be to check directly on the hosts (e.g., ip link cmd or whatever)

Reply:
Not easy to do a full check on this without getting into implementation details and/or excessive complications. I will try to bake something in.

Reply:
Just to clarify, there some other things that would surely fail during the process and that would be detected anyways.

Reply:
ok. my thought was that we could end up in a state where we set the MTU, and then read it with some oc command and
it gives back what we expect, but then in reality underneath it's not really changed. but, maybe those "some other things"
that would surely fail if they are not set like we expect is good enough to validate it. your call.

Reply:
Added some checks for node and pod MTU.